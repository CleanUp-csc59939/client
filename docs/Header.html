<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: Header</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: Header</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>Header<span class="signature">()</span><span class="type-signature"></span></h2>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    

    
    <h4 class="name" id="Header"><span class="type-signature"></span>new Header<span class="signature">()</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    <p>Header Component</p>
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="Shared_Header.js.html">Shared/Header.js</a>, <a href="Shared_Header.js.html#line17">line 17</a>
    </li></ul></dd>
    

    

    

    
</dl>




















    
    </div>

    

    

    

    

    

    

    

    
        <h3 class="subsection-title">Methods</h3>

        
            

    

    
    <h4 class="name" id="~CreateEvent"><span class="type-signature">(inner) </span>CreateEvent<span class="signature">()</span><span class="type-signature"> &rarr; {Component}</span></h4>
    

    



<div class="description">
    <p>Component for showing button to create an event</p>
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="Shared_Header.js.html">Shared/Header.js</a>, <a href="Shared_Header.js.html#line58">line 58</a>
    </li></ul></dd>
    

    

    

    
</dl>















<h5>Returns:</h5>

        
<div class="param-desc">
    <p>Returns the Create Event button</p>
</div>



<dl>
    <dt>
        Type
    </dt>
    <dd>
        
<span class="param-type">Component</span>


    </dd>
</dl>

    





        
            

    

    
    <h4 class="name" id="~onClickOut"><span class="type-signature">(inner) </span>onClickOut<span class="signature">(none)</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    <p>when user clicks outside of modal, it sets overlay in App.js and modal to false
which then hids the modal and removes opacity background in App.js</p>
</div>









    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>none</code></td>
            

            <td class="type">
            
            </td>

            

            

            <td class="description last"></td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="Shared_Header.js.html">Shared/Header.js</a>, <a href="Shared_Header.js.html#line31">line 31</a>
    </li></ul></dd>
    

    

    

    
</dl>















<h5>Returns:</h5>

        
<div class="param-desc">
    <p>none</p>
</div>



    





        
            

    

    
    <h4 class="name" id="~triggerModal"><span class="type-signature">(inner) </span>triggerModal<span class="signature">(input, refine)</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    <p>Sets overlay (in App.js) and showModal to true</p>
</div>









    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>input</code></td>
            

            <td class="type">
            
                
<span class="param-type">event</span>


            
            </td>

            

            

            <td class="description last"><p>read-only from search bar input</p></td>
        </tr>

    

        <tr>
            
                <td class="name"><code>refine</code></td>
            

            <td class="type">
            
                
<span class="param-type">function</span>


            
            </td>

            

            

            <td class="description last"><p>prop from connectSearchBox</p></td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="Shared_Header.js.html">Shared/Header.js</a>, <a href="Shared_Header.js.html#line42">line 42</a>
    </li></ul></dd>
    

    

    

    
</dl>




















        
    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="Banner.html">Banner</a></li><li><a href="CreateMeetUp.html">CreateMeetUp</a></li><li><a href="CustomSearch.html">CustomSearch</a></li><li><a href="Header.html">Header</a></li><li><a href="Home.html">Home</a></li><li><a href="Login.html">Login</a></li><li><a href="SingleEvent.html">SingleEvent</a></li></ul><h3>Global</h3><ul><li><a href="global.html#authHeader">authHeader</a></li><li><a href="global.html#ConvertDate">ConvertDate</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc/jsdoc">JSDoc 3.6.7</a> on Tue May 25 2021 00:59:11 GMT-0400 (Eastern Daylight Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>