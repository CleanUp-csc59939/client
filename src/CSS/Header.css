.site-page-header {
  border: 1px solid rgb(235, 237, 240);
}
.subtitle {
  padding: 0 2% 0 2%; /*horizontal padding*/
}

/*For web (s, m, l): Hide on xs screen*/
@media (max-width: 48em) {
  .web {
    display: none !important;
  }
}

/*For mobile (xs): Hide on small screen to large*/
@media (min-width: 48em) {
  .mobile {
    display: none !important;
  }
}

@media (min-width: 48em) and (max-width: 62em) {
  .hidden-sm {
    display: none !important;
  }
}
@media (min-width: 62em) and (max-width: 75em) {
  .hidden-md {
    display: none !important;
  }
}
@media (min-width: 75em) {
  .hidden-lg {
    display: none !important;
  }
}
