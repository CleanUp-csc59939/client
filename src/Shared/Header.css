.site-page-header {
  background-color: #bdc1f3ff;
  box-shadow: 0 0 20px 0 rgba(0, 0, 0, 0.2), 0 5px 5px 0 rgba(0, 0, 0, 0.24);
}

.homeButton {
  background-color: #4caf50;
  stroke: black;
  border: transparent;
}

.homeButton:hover {
  background-color: #228a26;
}

.headerTitle {
  color: whitesmoke;
}

.headerOther{
  color: whitesmoke;
}

.headerImg{
  justify-self: center;
  height: 75px;
  position: fixed;
  margin-top: -25px;
  margin-left: 37%;

}
.subtitle {
  padding: 0 2% 0 2%; /*horizontal padding*/
}

/*For web (s, m, l): Hide on xs screen*/
@media (max-width: 48em) {
  .web {
    display: none !important;
  }
}

/*For mobile (xs): Hide on small screen to large*/
@media (min-width: 48em) {
  .mobile {
    display: none !important;
  }
}

@media (min-width: 48em) and (max-width: 62em) {
  .hidden-sm {
    display: none !important;
  }
}
@media (min-width: 62em) and (max-width: 75em) {
  .hidden-md {
    display: none !important;
  }
}
@media (min-width: 75em) {
  .hidden-lg {
    display: none !important;
  }
}
