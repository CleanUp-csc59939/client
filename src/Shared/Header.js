import React, { useState } from 'react';
import { PageHeader, Button, Row, Col, Space, Input } from 'antd';
import { AiOutlinePlus } from 'react-icons/ai';
import AuthService from '../services/auth.service';
import { InstantSearch, Hits, connectSearchBox } from 'react-instantsearch-dom';
import { instantMeiliSearch } from '@meilisearch/instant-meilisearch';
import SearchHits from './SearchHits';
// import env from 'react-dotenv';
import './Header.css';

const ClickOutHandler = require('react-onclickout');

/**
 * Header Component
 * @component
 */
const Header = ({ currentUser /* pageTitle  */, setOverlay }) => {
  const [showModal, setShowModal] = useState(false);
  // logic to check if logged in goes here to switch between 2 different headers
  const logOut = () => {
    console.log('log out');
    AuthService.logout();
  };

  /**
   * when user clicks outside of modal, it sets overlay in App.js and modal to false
   * which then hids the modal and removes opacity background in App.js
   * @method
   * @param none
   * @return none
   */
  const onClickOut = () => {
    setOverlay(false);
    setShowModal(false);
  };

  /**
   * Sets overlay (in App.js) and showModal to true
   * @method
   * @param {event} input read-only from search bar input
   * @param {function} refine prop from connectSearchBox
   */
  const triggerModal = (event, refine) => {
    refine(event.currentTarget.value);
    setOverlay(true);
    setShowModal(true);
  };

  const searchClient = instantMeiliSearch('http://3.139.65.222/', 'NWFjZGNhMGZjMThjMDgzYjY4NTcyNGY1', {
    primaryKey: 'id',
  });

  /**
   * Component for showing button to create an event
   * @function
   * @return {Component} Returns the Create Event button
   *
   */
  const CreateEvent = () => {
    return (
      <a href='/myMeetUps/create' alt='' type='submit'>
        <div
          style={{
            height: 50,
            width: 200,
            backgroundColor: '#AEFFCF',
            borderRadius: 60,
            borderColor: '#AEFFCF',
          }}
        >
          <Row style={{ alignItems: 'center', justifyContent: 'center' }}>
            <Space size='small'>
              <h2 style={{ paddingTop: '5%' }}>Create Event</h2>
              <AiOutlinePlus size={20} style={{ paddingBottom: '2%' }} />
            </Space>
          </Row>
        </div>
      </a>
    );
  };
  if (currentUser) {
    return (
      <div className='site-page-header'>
        <img className='headerImg' src='/logo.png' alt='' />
        <Row className='web' style={{ padding: '2%' }}>
          <Col span={3} style={{ paddingTop: '10px' }}>
            <Button className='homeButton' type='link' href='/home'>
              <h2 style={{ fontWeight: 'bold', color: '#4F4F4F' }}>CleanUp</h2>
            </Button>
          </Col>
          <Col span={9}>
            <ClickOutHandler onClickOut={() => onClickOut()}>
              <InstantSearch indexName='events' searchClient={searchClient}>
                <CustomSearch triggerModal={triggerModal} />
                {showModal ? (
                  <div className='hits'>
                    <SearchHits hitComponent={Hits} />
                  </div>
                ) : null}
              </InstantSearch>
            </ClickOutHandler>
          </Col>
          <Col span={12}>
            <Row>
              <Space size='small'>
                <CreateEvent />
                <Col>
                  <Button type='link' href='/home'>
                    <h2>Home</h2>
                  </Button>
                </Col>
                <Col>
                  <Button type='link' href='/myMeetups'>
                    <h2>My Meet Ups</h2>
                  </Button>
                </Col>
                <Col>
                  <Button type='link' href='/profile'>
                    <h2>Profile</h2>
                  </Button>
                </Col>
                <Col>
                  <Button type='link' href='/login' onClick={logOut}>
                    <h2>Logout</h2>
                  </Button>
                </Col>
              </Space>
            </Row>
          </Col>
        </Row>
        <Row sm={4} className='mobile'>
          <Col span={8} offset={8}>
            <h2>Clean Up</h2>
          </Col>
        </Row>
      </div>
    );
  }
  return (
    <PageHeader
      className='site-page-header'
      ghost={false}
      title={[
        <Button className='homeButton' href='/home' key='1'>
          <h2 className='headerTitle'>CleanUp</h2>
        </Button>,
      ]}
      subTitle={[<img className='headerImg' src='./logo.png' alt='' />, "Let's Clean Up Our Home"]}
      extra={[
        <Button className='homeButton' href='/login' key='3'>
          <h4 className='headerOther'>Login</h4>
        </Button>,
        <Button className='homeButton' href='/signup' key='2'>
          <h4 className='headerOther'>SignUp</h4>
        </Button>,
      ]}
    ></PageHeader>
  );
};
export default Header;

/**
 * Custom search bar component wrapped in a instant-search helper function
 * @component
 * @param {function} triggerModal parent function to trigger search hits to show
 * @return                        search bar componennt
 *
 */
const CustomSearch = connectSearchBox(({ refine, currentRefinement, triggerModal }) => {
  return (
    <Input.Group style={{ paddingTop: '2%' }}>
      <Input
        value={currentRefinement}
        placeholder='Search Events'
        onChange={(event) => triggerModal(event, refine)}
        style={{ width: 400, borderRadius: 10 }}
      />
    </Input.Group>
  );
});
