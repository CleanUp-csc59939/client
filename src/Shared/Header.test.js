import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

const testUser = {
  userID: 15,
  name: null,
  bio: null,
  location: null,
  img: 'https://res.cloudinary.com/cleanup/image/upload/v1621121500/download_remvsp.png',
};

describe('Header', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Header currentUser={testUser} debug />);

    expect(component).toMatchSnapshot();
  });
});
