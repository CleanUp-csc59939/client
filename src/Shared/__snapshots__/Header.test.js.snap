// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render correctly in "debug" mode 1`] = `
<div
  className="site-page-header"
>
  <img
    alt=""
    className="headerImg"
    src="/logo.png"
  />
  <Row
    className="web"
    style={
      Object {
        "padding": "2%",
      }
    }
  >
    <Col
      span={3}
      style={
        Object {
          "paddingTop": "10px",
        }
      }
    >
      <Button
        className="homeButton"
        href="/home"
        type="link"
      >
        <h2
          style={
            Object {
              "color": "#4F4F4F",
              "fontWeight": "bold",
            }
          }
        >
          CleanUp
        </h2>
      </Button>
    </Col>
    <Col
      span={9}
    >
      <InstantSearch
        indexName="events"
        refresh={false}
        searchClient={
          Object {
            "client": MeiliSearch {
              "config": Object {
                "apiKey": "NWFjZGNhMGZjMThjMDgzYjY4NTcyNGY1",
                "host": "http://3.139.65.222/",
              },
              "httpRequest": HttpRequests {
                "headers": Object {
                  "Content-Type": "application/json",
                  "X-Meili-API-Key": "NWFjZGNhMGZjMThjMDgzYjY4NTcyNGY1",
                },
                "url": "http://3.139.65.222/",
              },
            },
            "getNumberPages": [Function],
            "hitsPerPage": 20,
            "page": 0,
            "paginateHits": [Function],
            "paginationTotalHits": 200,
            "placeholderSearch": true,
            "primaryKey": "id",
            "search": [Function],
            "transformToISHits": [Function],
            "transformToISResponse": [Function],
            "transformToMeiliSearchParams": [Function],
          }
        }
        stalledSearchDelay={200}
      >
        <ConnectorWrapper />
        <ConnectorWrapper
          hitComponent={[Function]}
        />
      </InstantSearch>
    </Col>
    <Col
      span={12}
    >
      <Row>
        <Space
          size="small"
        >
          <CreateEvent />
          <Col>
            <Button
              href="/home"
              type="link"
            >
              <h2>
                Home
              </h2>
            </Button>
          </Col>
          <Col>
            <Button
              href="/myMeetups"
              type="link"
            >
              <h2>
                My Meet Ups
              </h2>
            </Button>
          </Col>
          <Col>
            <Button
              href="/profile"
              type="link"
            >
              <h2>
                Profile
              </h2>
            </Button>
          </Col>
          <Col>
            <Button
              href="/login"
              onClick={[Function]}
              type="link"
            >
              <h2>
                Logout
              </h2>
            </Button>
          </Col>
        </Space>
      </Row>
    </Col>
  </Row>
  <Row
    className="mobile"
    sm={4}
  >
    <Col
      offset={8}
      span={8}
    >
      <h2>
        Clean Up
      </h2>
    </Col>
  </Row>
</div>
`;
